app/src/main/java/ch/epfl/sdp/utils/DependencyFactory.java

```diff
@@ -0,0 +1,77 @@
+package ch.epfl.sdp.utils;
+
+import androidx.annotation.VisibleForTesting;
+
+import ch.epfl.sdp.database.firebase.api.ClientDatabaseAPI;
+import ch.epfl.sdp.database.firebase.api.ClientFirestoreDatabaseAPI;
+import ch.epfl.sdp.database.firebase.api.CommonDatabaseAPI;
+import ch.epfl.sdp.database.firebase.api.CommonFirestoreDatabaseAPI;
+import ch.epfl.sdp.database.firebase.api.ServerDatabaseAPI;
+import ch.epfl.sdp.database.firebase.api.ServerFirestoreDatabaseAPI;
+import ch.epfl.sdp.login.AuthenticationAPI;
+import ch.epfl.sdp.login.FirebaseAuthenticationAPI;
+
+public class DependencyFactory {
```

Comment:
That seem to be a cool class! Should not it be in the tests?

Reply:
we should keep it as it is because this part is used from some activity.